src/TensorFlowNET.Keras/Engine/Model.Fit.cs

```diff
@@ -40,11 +42,17 @@ public partial class Model
                 throw new InvalidArgumentError(
                     $"The array x and y should have same value at dim 0, but got {x.dims[0]} and {y.dims[0]}");
             }
-            int train_count = Convert.ToInt32(x.dims[0] * (1 - validation_split));
-            var train_x = x[new Slice(0, train_count)];
-            var train_y = y[new Slice(0, train_count)];
-            var val_x = x[new Slice(train_count)];
-            var val_y = y[new Slice(train_count)];
+
+            var train_x = x;
+            var train_y = y;
+
+            if (validation_split != 0f && validation_data == null)
```

Comment:
When validation_split is not equals to 0 and validation_data is not null, we ignore the validation_split.